### PR TITLE
Quicksettings version 3

### PR DESCRIPTION
**Pull request by MatthijsKamstra**
Hi Mark, 

the changes are not that big for version 3. 
The only thing I "struggle" with is the getter/setter `getValue / setValue` because of the Dynamic return type.
Solution would be to bring back the "old" specific getter/setters

Matthijs

### CODE REVIEW

**Comment by MatthijsKamstra** [comment]
Ah tests... I will wait for your decision, because we could just keep the getter setters.... 😄

File: example/bin/index.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <html lang="en">
 	<head>
 		<title>hx-quicksettings example </title>
-		<script type="text/javascript" src="https://cdn.jsdelivr.net/quicksettings/2.2.1/quicksettings.min.js"></script>
+		<script type="text/javascript" src="https://cdn.jsdelivr.net/quicksettings/latest/quicksettings.min.js"></script>
```
Comment (markknol) [inline example/bin/index.html]: Can you use an explicit version here? This is nicer for releases/tags.

File: example/src/Main.hx [hunk 2]
```diff
@@ -62,33 +66,35 @@ class Main
...
-				"title": "text area test",
-				"value": "just some text"
-			},{
-				"type": "boolean",
-				"title": "boolean test",
-				"value": true
-			}]
-		}';
-		var panel6 = QuickSettings.parse(json);
+
```
Comment (markknol) [inline example/src/Main.hx]: I think you can just remove the old stuff, no need keep it in comments (That's why we use source control 😄)

File: quicksettings/QuickSettings.hx [hunk 3]
```diff
@@ -147,35 +141,57 @@ extern class QuickSettings
...
+	public function saveInLocalStorage (name:String):QuickSettings;
+
+	/**
+        Clears any saved values in local storage.
+    **/
+	public function clearLocalStorage (name:String):QuickSettings;
+
+
+	/**
+		This is the new getter/setter for QuickSettings (version 3).
```
Comment (markknol) [inline quicksettings/QuickSettings.hx]: Can you replace the comments to: "Sets value of the control with the given title."

Btw according the original source code it still returns `QuickSettings`

File: quicksettings/QuickSettings.hx [hunk 4]
```diff
@@ -331,38 +359,38 @@ extern class QuickSettings
 typedef QSettings =
 {
 	/** optional string, default "QuickSettings" **/
-	@optional var title : String;
+	@:optional var title : String;
```
Comment (markknol) [inline quicksettings/QuickSettings.hx:314]: Is this an "oopsie"?

**Review by markknol — COMMENTED** [review]
Hey I missed this PR, sorry. Can you check the following things? (and fix the tests? 😛 )

File: quicksettings/QuickSettings.hx (hunk 4 above)
Comment (MatthijsKamstra) [reply quicksettings/QuickSettings.hx:314]: this is correct, the previous version was an oepsie
(https://haxe.org/manual/type-system-typedef.html#optional-fields)

**Comment by MatthijsKamstra** [comment]
I think I am correct that the only test it does, if it builds... then I have fixed it!

**Comment by MatthijsKamstra** [comment]
still save to do a pull request

**Comment by markknol** [comment]
Thanks, merged and released as new version.

**Pull request merged**